### PR TITLE
refactor: name change of FTF -> GBTS seeding

### PR DESCRIPTION
**Pull request by Rosie-Hasan**
Change naming of new seeding algorithm from FTF (FastTrackFinder) to GBTS (Graph Based Track Seeding) as this is a clearer desription of the algorithm. Name changes disccuessed in issue #2831 .  
Code implemented in PRs #2227  #2726  and more details given presented at [ACTS-ITK presention ](https://indico.cern.ch/event/1321208/#20-update-on-ftf-implementatio).

**Name changes**: 
SeedFinderFTF -> SeedFinderGBTS
GNN_DataStorage -> GBTSDataStorage
GNN_Geometry -> GBTSGeometry
SeedingFTFAlgorithm -> GBTSSeedingAlgorithm 
FasTrackConnector-> GBTSConnector
TrigBase -> GBTSBase
full_chain_itk_FTF.py -> full_chain_itk_GBTS.py

### CODE REVIEW

File: Core/include/Acts/Seeding/GbtsTrackingFilter.hpp [hunk 1]
```diff
@@ -285,6 +283,21 @@ class TrigFTF_GNN_TrackingFilter {
...
+              << std::endl;
+    std::cout << "  printing pre  arithmetic error 11 : " << ts.m_Cx[1][1]
+              << std::endl;
+    std::cout << "  printing pre  arithmetic error 12 : " << ts.m_Cx[1][2]
+              << std::endl;
+    std::cout << "  printing pre  arithmetic error 22 : " << ts.m_Cx[2][2]
+              << std::endl;
+    std::cout << " printing pre  arithmetic error A : " << A << std::endl;
+    std::cout << " printing pre  arithmetic error B : " << B << std::endl;
+
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/GbtsTrackingFilter.hpp:300]: I think, this not part of this PR.

File: Core/include/Acts/Seeding/GbtsTrackingFilter.hpp [hunk 2]
```diff
@@ -305,7 +318,8 @@ class TrigFTF_GNN_TrackingFilter {
     Cy[0][1] = Cy[1][0] = ts.m_Cy[0][1] + dr * ts.m_Cy[1][1];
     Cy[1][1] = ts.m_Cy[1][1];
 
-    // chi2 test
+    // std::cout << " printing after where arithmetic error is " << Cx[0][0] <<
+    // std::endl ; chi2 test
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/GbtsTrackingFilter.hpp:322]: This as well.

File: Core/include/Acts/Seeding/SeedFinderGbts.hpp [hunk 3]
```diff
@@ -51,47 +51,47 @@ class SeedFinderFTF {
   //   using tree_t = KDTree<NDims, internal_sp_t *, ActsScalar, std::array, 4>;
 
   // constructors
-  SeedFinderFTF(const SeedFinderFTFConfig<external_spacepoint_t> &config,
-                const TrigFTF_GNN_Geometry<external_spacepoint_t> &GNNgeo);
+  SeedFinderGbts(const SeedFinderGbtsConfig<external_spacepoint_t> &config,
+                 const GbtsGeometry<external_spacepoint_t> &Gbtsgeo);
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderGbts.hpp:55]:
```suggestion
  SeedFinderGbts(const SeedFinderGbtsConfig<external_spacepoint_t> &config,
                 const GbtsGeometry<external_spacepoint_t> &gbtsGeo);
```

For variable names we should start with lower case. (Also some occurrences below)

File: Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp [hunk 4]
```diff
@@ -71,9 +71,9 @@ class SeedingFTFAlgorithm final : public IAlgorithm {
 
   // own class functions
   // make the map
-  std::map<std::pair<int, int>, std::pair<int, int>> Make_ACTS_FTF_Map() const;
+  std::map<std::pair<int, int>, std::pair<int, int>> Make_ACTS_Gbts_Map() const;
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp:74]:
```suggestion
  std::map<std::pair<int, int>, std::pair<int, int>> makeActsGbtsMap() const;
```
Similar below.

File: Core/include/Acts/Seeding/SeedFinderGbts.ipp [hunk 5]
```diff
@@ -530,17 +524,18 @@ void SeedFinderFTF<external_spacepoint_t>::runGNN_TrackFinder(
...
-             reverse_iterator sIt = rs.m_vs.rbegin();
+    for (typename std::vector<
+             Acts::GbtsEdge<external_spacepoint_t>*>::reverse_iterator sIt =
+             rs.m_vs.rbegin();
          sIt != rs.m_vs.rend(); ++sIt) {
       (*sIt)->m_level = -1;  // mark as collected
 
       if (sIt == rs.m_vs.rbegin()) {
-        vSP.push_back(&(*sIt)->m_n1->m_sp_FTF);
+        vSP.push_back(&(*sIt)->m_n1->m_sp_Gbts);
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderGbts.ipp:536]:
```suggestion
        vSP.push_back(&(*sIt)->m_n1->m_spGbts);
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]
Looks already quite fine. I think, there are only(?) 3 things to change:
- Start variable names with lower case
- Drop most of the underscores except for `m_afterThatNoUnderScores` and maybe in the python scripts
- Change `*ipp` to `.*cpp` when you are already renaming files. But this is argueable.

**Comment by Rosie-Hasan** [comment]
@AJPfleger Thank you for your help, I am making the name changes for the variables. For the ipp -> cpp I just wanted to double check as all the other files in core seeding are ipp/hpp

**Comment by AJPfleger** [comment]
> For the ipp -> cpp I just wanted to double check as all the other files in core seeding are ipp/hpp
I see, good spot. Then let's keep them as they are :)

File: Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp [hunk 6]
```diff
@@ -49,7 +49,7 @@ class SeedingFTFAlgorithm final : public IAlgorithm {
 
     std::shared_ptr<const Acts::TrackingGeometry> trackingGeometry;
 
-    std::map<std::pair<int, int>, std::pair<int, int>> ACTS_FTF_Map;
+    std::map<std::pair<int, int>, std::pair<int, int>> ACTSGbtsMap;
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp:52]:
```suggestion
    std::map<std::pair<int, int>, std::pair<int, int>> ActsGbtsMap;
```

File: Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp [hunk 7]
```diff
@@ -71,9 +71,9 @@ class SeedingFTFAlgorithm final : public IAlgorithm {
 
   // own class functions
   // make the map
-  std::map<std::pair<int, int>, std::pair<int, int>> Make_ACTS_FTF_Map() const;
+  std::map<std::pair<int, int>, std::pair<int, int>> makeACTSGbtsMap() const;
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/GbtsSeedingAlgorithm.hpp:74]:
```suggestion
  std::map<std::pair<int, int>, std::pair<int, int>> makeActsGbtsMap() const;
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 8]
```diff
@@ -71,43 +71,43 @@ ActsExamples::SeedingFTFAlgorithm::SeedingFTFAlgorithm(
...
-  // fastrack
-  std::unique_ptr<Acts::FasTrackConnector> input_fastrack =
-      std::make_unique<Acts::FasTrackConnector>(input_ifstream);
+  // connector
+  std::unique_ptr<Acts::GbtsConnector> inputConnector =
+      std::make_unique<Acts::GbtsConnector>(input_ifstream);
 
-  mGNNgeo = std::make_unique<Acts::TrigFTF_GNN_Geometry<SimSpacePoint>>(
-      m_cfg.seedFinderConfig.m_layerGeometry, input_fastrack);
+  mGbtsgeo = std::make_unique<Acts::GbtsGeometry<SimSpacePoint>>(
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:85]:
```suggestion
  m_gbtsGeo = std::make_unique<Acts::GbtsGeometry<SimSpacePoint>>(
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 9]
```diff
@@ -118,7 +118,7 @@ ActsExamples::ProcessCode ActsExamples::SeedingFTFAlgorithm::execute(
   }
 
   // this is now calling on a core algorithm
-  Acts::SeedFinderFTF<SimSpacePoint> finder(m_cfg.seedFinderConfig, *mGNNgeo);
+  Acts::SeedFinderGbts<SimSpacePoint> finder(m_cfg.seedFinderConfig, *mGbtsgeo);
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:121]:
```suggestion
  Acts::SeedFinderGbts<SimSpacePoint> finder(m_cfg.seedFinderConfig, *m_gbtsGeo);
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 10]
```diff
@@ -130,28 +130,28 @@ ActsExamples::ProcessCode ActsExamples::SeedingFTFAlgorithm::execute(
...
+  // trigGbts file :
   Acts::RoiDescriptor internalRoi(0, -4.5, 4.5, 0, -M_PI, M_PI, 0, -150.0,
                                   150.0);
   // ROI file:
   //  Acts::RoiDescriptor internalRoi(0, -5, 5, 0, -M_PI, M_PI, 0, -225.0,
   //                                  225.0);
 
   // new version returns seeds
-  SimSeedContainer seeds = finder.createSeeds(internalRoi, *mGNNgeo);
+  SimSeedContainer seeds = finder.createSeeds(internalRoi, *mGbtsgeo);
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:143]:
```suggestion
  SimSeedContainer seeds = finder.createSeeds(internalRoi, *m_gbtsGeo);
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 11]
```diff
@@ -71,43 +71,43 @@ ActsExamples::SeedingFTFAlgorithm::SeedingFTFAlgorithm(
           Acts::SeedFilter<SimSpacePoint>(m_cfg.seedFilterConfig));
 
   // map
-  m_cfg.ACTS_FTF_Map = Make_ACTS_FTF_Map();
+  m_cfg.ACTSGbtsMap = makeACTSGbtsMap();
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:74]:
```suggestion
  m_cfg.ActsGbtsMap = makeActsGbtsMap();
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 12]
```diff
@@ -164,28 +164,28 @@ ActsExamples::SeedingFTFAlgorithm::Make_ACTS_FTF_Map() const {
...
+std::vector<Acts::GbtsSP<ActsExamples::SimSpacePoint>>
+ActsExamples::GbtsSeedingAlgorithm::MakeGbtsSpacePoints(
     const AlgorithmContext &ctx,
     std::map<std::pair<int, int>, std::pair<int, int>> map) const {
   // create space point vectors
   std::vector<const ActsExamples::SimSpacePoint *> spacePoints;
-  std::vector<Acts::FTF_SP<ActsExamples::SimSpacePoint>> FTF_spacePoints;
-  FTF_spacePoints.reserve(
+  std::vector<Acts::GbtsSP<ActsExamples::SimSpacePoint>> GbtsSpacePoints;
+  GbtsSpacePoints.reserve(
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:188]:
```suggestion
  std::vector<Acts::GbtsSP<ActsExamples::SimSpacePoint>> gbtsSpacePoints;
  gbtsSpacePoints.reserve(
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 13]
```diff
@@ -232,36 +232,36 @@ ActsExamples::SeedingFTFAlgorithm::Make_FTF_spacePoints(
...
-      FTF_spacePoints.emplace_back(&spacePoint, FTF_id, combined_id);
+      // fill Gbts vector with current sapce point and ID
+      GbtsSpacePoints.emplace_back(&spacePoint, Gbts_id, combined_id);
     }
   }
-  ACTS_VERBOSE("Space points successfully assigned FTF ID");
+  ACTS_VERBOSE("Space points successfully assigned Gbts ID");
 
-  return FTF_spacePoints;
+  return GbtsSpacePoints;
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:260]:
```suggestion
  return gbtsSpacePoints;
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 14]
```diff
@@ -294,29 +294,29 @@ ActsExamples::SeedingFTFAlgorithm::LayerNumbering() const {
...
                        0);  // here the key needs to be pair of(vol*100+lay, 0)
-    auto Find = m_cfg.ACTS_FTF_Map.find(key);
-    int FTF_id = 0;               // initialise first to avoid FLTUND later
-    FTF_id = Find->second.first;  // new map, item is pair want first
+    auto Find = m_cfg.ACTSGbtsMap.find(key);
+    int Gbts_id = 0;               // initialise first to avoid FLTUND later
+    Gbts_id = Find->second.first;  // new map, item is pair want first
     if (Find ==
-        m_cfg.ACTS_FTF_Map
+        m_cfg.ACTSGbtsMap
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:306]:
```suggestion
        m_cfg.ActsGbtsMap
```

File: Examples/Python/src/TrackFinding.cpp [hunk 15]
```diff
@@ -273,10 +273,10 @@ void addTrackFinding(Context& ctx) {
                                 seedFinderOptions);
 
   ACTS_PYTHON_DECLARE_ALGORITHM(
-      ActsExamples::SeedingFTFAlgorithm, mex, "SeedingFTFAlgorithm",
+      ActsExamples::GbtsSeedingAlgorithm, mex, "GbtsSeedingAlgorithm",
       inputSpacePoints, outputSeeds, seedFilterConfig, seedFinderConfig,
       seedFinderOptions, layerMappingFile, geometrySelection, inputSourceLinks,
-      trackingGeometry, ACTS_FTF_Map, fill_module_csv, inputClusters);
+      trackingGeometry, ACTSGbtsMap, fill_module_csv, inputClusters);
```
Comment (AJPfleger) [inline Examples/Python/src/TrackFinding.cpp:279]:
```suggestion
      trackingGeometry, ActsGbtsMap, fill_module_csv, inputClusters);
```

File: Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp [hunk 16]
```diff
@@ -232,36 +232,36 @@ ActsExamples::SeedingFTFAlgorithm::Make_FTF_spacePoints(
...
 
       // access IDs from map
       int eta_mod = Find->second.second;
-      int combined_id = FTF_id * 1000 + eta_mod;
+      int combined_id = Gbts_id * 1000 + eta_mod;
 
-      // fill FTF vector with current sapce point and ID
-      FTF_spacePoints.emplace_back(&spacePoint, FTF_id, combined_id);
+      // fill Gbts vector with current sapce point and ID
+      GbtsSpacePoints.emplace_back(&spacePoint, Gbts_id, combined_id);
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/GbtsSeedingAlgorithm.cpp:255]:
```suggestion
      gbtsSpacePoints.emplace_back(&spacePoint, Gbts_id, combined_id);
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]
I think after these changes we are ready to go.

There are still a few other variables that need to be renamed, but they are not really in the scope of this PR. Since you planned to refactor/modernise the code code, some of them might even vanish. So no need to put too much thinking into it right now.

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [53bcd7462d688f7db6ef2fa47301a6ed1b681d5d]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35130095) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6708874)!

**Pull request merged**